protocol/src/main/java/tw/waterball/survive/io/protocol/scanner/ModuleClassesScanner.java

```diff
@@ -0,0 +1,69 @@
+package tw.waterball.survive.io.protocol.scanner;
+
+import java.io.File;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.*;
+
+/**
+ * @author - wally55077@gmail.com
+ */
+
+public final class ModuleClassesScanner {
+
+    private ModuleClassesScanner() {
+
+    }
+
+    /**
+     * Give a package name to this method ,
+     * it would scan whole module and returns all of Classes with a Set.
+     **/
+    public static Set<Class<?>> scanClassesByPackageName(String packageName) {
+        Set<Class<?>> classes = new HashSet<>();
+        String packageDirName = packageName.replace('.', '/');
+        try {
+            Enumeration<URL> dirs = Thread.currentThread().getContextClassLoader().getResources(packageDirName);
+            while (dirs.hasMoreElements()) {
+                URL url = dirs.nextElement();
+                String protocol = url.getProtocol();
+                if ("file".equals(protocol)) {
+                    String filePath = URLDecoder.decode(url.getFile(), "UTF-8");
+                    findClassesInPackageOnFileSystem(packageName, filePath, classes);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return classes;
+    }
+
+    private static void findClassesInPackageOnFileSystem(String packageName, String packagePath, Set<Class<?>> classes) {
+        File directory = new File(packagePath);
+        if (directory.exists() && directory.isDirectory()) {
+            File[] dirFiles = directory.listFiles();
+            Arrays.stream(Objects.requireNonNull(dirFiles))
+                    .parallel()
+                    .forEach(file -> findClassesInDirectory(packageName, classes, file, file.getName()));
+        }
+    }
+
+    private static void findClassesInDirectory(String packageName, Set<Class<?>> classes, File file, String fileName) {
+        if (file.isDirectory()) {
+            findClassesInPackageOnFileSystem(packageName + "." + fileName, file.getAbsolutePath(), classes);
+        } else {
+            findClassByClassName(packageName, classes, fileName);
+        }
+    }
+
+    private static void findClassByClassName(String packageName, Set<Class<?>> classes, String fileName) {
+        try {
+            String className = fileName.substring(0, fileName.length() - 6);
+            String path = packageName + '.' + className;
+            classes.add(Thread.currentThread().getContextClassLoader().loadClass(path));
+        } catch (ClassNotFoundException e) {
+            System.out.println("添加用戶自定義視圖類錯誤找不到此類的.class文件");
```

Comment:
Use English, and don't use `System.out.println` to log the error, you can throw `RuntimeException` instead.